### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by saraialvarezrosas**

### CODE REVIEW

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java [hunk 1]
```diff
@@ -89,7 +90,9 @@ public static void faceDetection (String file, String type) {
                         new Point(rect.x + rect.width, rect.y + rect.height) , new Scalar(0, 0, 255), 3);
             }
             String finalFile = file.substring(22,file.length());
-            Imgcodecs.imwrite("D:\\workspacejala\\progra102\\JU-OBJECT-RECOGNITION\\searchobject\\src\\main\\resources\\images\\"+ finalFile, src);
+            Imgcodecs.imwrite(currentDir + "\\src\\main\\resources\\images\\" + finalFile, src);
+            System.out.println("la imagen   :  "+ currentDir + "\\src\\main\\resources\\images\\" + finalFile);
```
Comment (DaniMB95) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java]: delete the system.out.println or replace it with a logger

Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java]: change the System.out.println with logger

**Review by DaniMB95 — APPROVED** [review]

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * Filters are Java classes that implement the javax.servlet.Filter interface,
+ * and whose mission is to intercept requests before they reach the servlets
+ * and after, to perform certain operations.
+ * @author Sarai Alvarez
+ * @version 1.0
+ */
+@WebFilter(urlPatterns = "/emotionRecognition")
+
+public class EmotionRecognitionMiddleware implements Filter {
+    private static final Logger LOG = new At18Logger().getLogger();
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java:28]: The metod `.getLogger()` is a static method, it couldn't be accesed in a static way `At18Logger.getLogger();` ?

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     * @param //request it's the image that user will upload to analyze
+     * @param //response the type of face that we want to detect
+     */
+    @Override
+    public void doFilter(ServletRequest request,
+                         ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException
+    {
+        HttpServletRequest req = (HttpServletRequest) request; //
+        HttpServletResponse res = (HttpServletResponse) response;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java]: Is this variable `res` being used? if it's not the case, please remove it

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Filters are Java classes that implement the javax.servlet.Filter interface,
+ * and whose mission is to intercept requests before they reach the servlets
+ * and after, to perform certain operations.
+ * @author Sarai Alvarez
+ * @version 1.0
+ */
+@WebFilter(urlPatterns = "/faceDetection")
+public class FaceDetectionControllerMiddleware implements Filter {
+    private static final Logger LOG = new At18Logger().getLogger();
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java:27]: The metod `.getLogger()` is a static method, it couldn't be accesed in a static way `At18Logger.getLogger();` ?

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+/**
+ * Filters are Java classes that implement the javax.servlet.Filter interface,
+ * and whose mission is to intercept requests before they reach the servlets
+ * and after, to perform certain operations.
+ * @author Sarai Alvarez
+ * @version 1.0
+ */
+@WebFilter(urlPatterns = "/processFrame")
+public class ProcessFrameControllerMidleware implements Filter {
+    private static final Logger LOG = new At18Logger().getLogger();
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java:27]: The metod `.getLogger()` is a static method, it couldn't be accesed in a static way `At18Logger.getLogger();` ?

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+     * @param //request it's the image that user will upload to analyze
+     * @param //response the type of face that we want to detect
+     */
+    @Override
+    public void doFilter(ServletRequest request,
+                         ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException
+    {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse res = (HttpServletResponse) response;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java]: Is this variable `res` being used? if it's not the case, please remove it

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+     * @param //request it's the image that user will upload to analyze
+     * @param //response the type of face that we want to detect
+     */
+    @Override
+    public void doFilter(ServletRequest request,
+                         ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException
+    {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse res = (HttpServletResponse) response;
```
Comment (rodri-98) [inline searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java]: Is this variable `res` being used? if it's not the case, please remove it

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java (hunk 3 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java]: The variable is not being used, it was removed.

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java (hunk 6 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java]: The variable is not being used, it was removed.

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java (hunk 7 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java]: The variable is not being used, it was removed.

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java (hunk 2 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/EmotionRecognitionMiddleware.java:28]: Is necessary keep it and cannot be removed, must be initialized.

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java (hunk 4 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java:27]: Is necessary keep it and cannot be removed, must be initialized.

File: searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java (hunk 5 above)
Comment (saraialvarezrosas) [reply searchobject/src/main/java/org/jalau/at18/searchobject/middleware/ProcessFrameControllerMidleware.java:27]: Is necessary keep it and cannot be removed, must be initialized.

**Review by rodri-98 — APPROVED** [review]
Nice job !

**Review by ferAguilar02 — APPROVED** [review]

**Pull request merged**